core/block_svg.js

```diff
@@ -272,27 +272,24 @@ Blockly.BlockSvg.prototype.getIcons = function() {
  * @param {Blockly.BlockSvg} newParent New parent block.
  */
 Blockly.BlockSvg.prototype.setParent = function(newParent) {
-  if (newParent == this.parentBlock_) {
+  var oldParent = this.parentBlock_;
```

Comment:
this is a totally separate optimization here (setParent) that will get landed separately - ignore for now, it's just a base difference (i'm on "performance")